src/Bicep.Core/Emit/ParametersFileTemplateWriter.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+
+namespace Bicep.Core.Emit
+{
+    public class ParametersFileTemplateWriter : TemplateWriter
+    {
+        public ParametersFileTemplateWriter(SemanticModel semanticModel, EmitterSettings settings): base(semanticModel, settings)
+        {
+            this.context = new EmitterContext(semanticModel, settings);
+            this.settings = settings;
+        }
+
+        private readonly EmitterContext context;
+        private readonly EmitterSettings settings;
+
+        public void Write(JsonTextWriter writer, string existingContent)
+        {
+            // Template is used for calcualting template hash, template jtoken is used for writing to file.
+            var templateJToken = GenerateTemplate();
+            if (templateJToken.SelectToken(GeneratorMetadataPath) is not JObject generatorObject)
+            {
+                throw new InvalidOperationException($"generated template doesn't contain a generator object at the path {GeneratorMetadataPath}");
+            }
+
+            if (!string.IsNullOrWhiteSpace(existingContent))
```

Comment:
Does the user get a reasonable error experience if the parse fails?

Reply:
current infrastructure to handle errors will catch the exception and handle it properly. I based off this code from the `TemplateWriter` class
 https://github.com/polatengin/bicep/blob/dbe1bff5efa6a38419a9330c7e54cae85f84dfc1/src/Bicep.Core/Emit/TemplateWriter.cs#L104